6-pubsub/package.json

```diff
@@ -8,8 +8,9 @@
     "start": "node ${SCRIPT:-app.js}",
     "monitor": "nodemon ${SCRIPT:-app.js}",
     "deploy": "gcloud preview app deploy app.yaml worker.yaml",
-    "lint": "jshint --exclude-path=.gitignore .",
-    "test": "npm run lint",
+    "lint": "jshint --exclude-path=../.gitignore .",
+    "mocha": "mocha -R dot test/unit/*.test.js",
```

Comment:
> will this run mocha from the installed dev dependency?

Yes

For all dependencies installed via `npm install` that have a `bin` specified in their `package.json` file, npm creates a symlink to the specified binary and puts the symlink in `node_modules/.bin`. `npm run` places `node_modules/.bin` in the `PATH` when it executes, so `npm run mocha` executes `mocha -R dot test/unit/*.test.js`, which finds `mocha` in the PATH thanks to npm placing a symlink to the mocha binary in `node_modules/.bin`.

`"mocha": "mocha -R dot test/unit/*.test.js"` is the same as
`"mocha": "node_modules/.bin/mocha -R dot test/unit/*.test.js"` is the same as
`"mocha": "node_modules/mocha/bin/mocha -R dot test/unit/*.test.js"`

https://www.google.com/url?sa=D&q=https%3A%2F%2Fdocs.npmjs.com%2Fcli%2Frun-script